docs/enhancements/arbiter-clusters.md

```diff
@@ -0,0 +1,144 @@
+
+### Implementation Details/Notes/Constraints
+
+The enums for cluster's hyperthreading and disk encryption will be:
+- 'none'
+- 'masters'
+- 'arbiters'
+- 'workers'
+- 'masters,arbiters'
+- 'masters,workers'
+- 'arbiters,workers'
+- 'masters,arbiters,workers'
+- 'all' which will be the same as 'masters,arbiters,workers'
+
+When generating the ignitions for a TNA cluster, a new base ignition named
+arbiter.ign will be created. If it exists we need to patch it and upload it
+to s3. The list of patches arbiter.ign needs:
+- Set CA cert if it was provided.
+- If all the hosts in the cluster have IPv6 address then we need to add a
+  NetworkManager configuration ensuring that IPv6 DHCP requests use consistent
+  client identification.
+
+Our validations will need to be updated as follows:
+- A host can only be assigned the arbiter role if the cluster is TNA.
+- A cluster's arbiter_count must be zero or higher.
+- A cluster's number of arbiter nodes must be exactly the cluster's
+  arbiter_count, in order for the cluster to be ready for installation.
+- TNA clusters' can set control_plane_count to be 2.
+- TNA clusters' platform must be baremetal.
+- TNA clusters are expected to be TP stating from OCP 4.19.
+
+In the cluster's transition handler, the function enoughMastersAndWorkers
+needs to be updated to also check for arbiters. This is needed to correctly
+decide if the cluster is in installing/finalizing.
+
+We need to add arbiter's hardware requirements in:
+- The HW_VALIDATOR_REQUIREMENTS environment variable (by default we'll take
+  the worker's requirements to avoid issues mid-development).
+- The HostTypeHardwareRequirementsWrapper structs used by the operators.
+
+We will auto-assign the arbiter role to a host if all the following are true:
+- The host's cluster has less arbiter nodes than the cluster's arbiter_count. 
+- The host has the minimum hardware requirements for arbiter nodes.
+- Either the host doesn't have the minimum hardware requirements for master
+  nodes, or the host's cluster has enough master nodes.
+
+Outside of assisted-service, we also need to do the following:
+- After we update the API in assisted-services, we need to update the vendor
+  directory in assisted-installer-agent.
+- We need to update the function waitForMinMasterNodes in assisted-installer
+  to set the min master nodes to be the minimum of 2 (the current value)
+  and number of control planes (read it from the install-config) minus 1.
+
+Until we have this feature fully implemented, we should have an environment
+variable to decide if we allow TNA clusters (default value is false).
+
+### Risks and Mitigations
+
+N/A
+
+## Design Details
+
+### Open Questions
+
+N/A
+
+### UI Impact
+
+- In the host discovery we need to allow setting a host's role to be arbiter.
+- In the cluster details:
+  - we need to allow setting the arbiter_count, with zero as the default.
+  - We need to allow setting control_plane_count to 2, but then force setting
```

Comment:
@carbonin Will this be correct when we have the Two Node OpenShift with Fencing?

Reply:
Right, I think this is fine for now when we only have TNA, but I don't think there's anything better we can do until we also need to handle TNF.

Reply:
I suppose I take this back based on the conversation in the kickoff.

It might be better to just validate that we have a good combination rather than forcing arbiters to 1 since that's something we'd need to pull out once we have TNF.